### PR TITLE
CIF-2227 - add missing image alt attribute

### PR DESCRIPTION
**Pull request by mhaack**
## Description

* add missing image alt attribute for product teaser component
* add missing image alt attribute for product collection and derived components

## Related Issue

CIF-2227

## How Has This Been Tested?

Extended unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/test/java/com/adobe/cq/commerce/core/search/internal/converters/ProductToProductListItemConverterTest.java [hunk 1]
```diff
@@ -72,6 +73,7 @@ public void setUp() {
         when(productInterface.getUrlKey()).thenReturn(URL);
         when(productInterface.getName()).thenReturn(PRODUCT_NAME);
         when(productInterface.getSmallImage().getUrl()).thenReturn(IMAGE_URL);
+        when(productInterface.getSmallImage().getLabel()).thenReturn(IMAGE_LABEL);
```
Comment (Buuhuu) [inline bundles/core/src/test/java/com/adobe/cq/commerce/core/search/internal/converters/ProductToProductListItemConverterTest.java:76]: you should assert the mocked value.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java [hunk 2]
```diff
@@ -51,6 +51,16 @@ default String getImageURL() {
...
     }
 
+    /**
+     * Returns the product image alt text of this {@code ProductListItem}.
+     *
+     * @return the product image alt text of this list item or {@code null}
+     */
+    @Nullable
+    default String getImageAlt() {
+        throw new UnsupportedOperationException();
```
Comment (LSantha) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java:61]: I think we stopped doing this a while ago.

File: bundles/core/src/test/java/com/adobe/cq/commerce/core/search/internal/converters/ProductToProductListItemConverterTest.java (hunk 1 above)
Comment (mhaack) [reply bundles/core/src/test/java/com/adobe/cq/commerce/core/search/internal/converters/ProductToProductListItemConverterTest.java:79]: Done

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java (hunk 2 above)
Comment (mhaack) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java:61]: I followed the other methods if the interface, but I'm also ok to remove it. Maybe we should clean up all methods and other classes?

Comment (LSantha) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java:61]: +1 for cleanup. 
There was already a cleanup in #199. A quick search shows that this is the only Sling model interface with such methods but there are new occurrences in datalayer like CategoryData, ProductData. @laurentiumagureanu , is that OK to clean them up as well?

Comment (mhaack) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/common/ProductListItem.java]: Removed defaults for the entire ProductListItem interface.

**Review by herzog31 — APPROVED** [review]

**Pull request merged**